### ISSUE TITLE
Allow setting the GitHub status context in .taskcluster.yml

### ISSUE THREAD

**Issue by jgraham** [taskcluster/taskcluster-github#270]
For wpt we have a situation where on some pushes we run multiple TaskCluster taskgroups with different settings (this is controlled by pushes to different branches). Unfortunately the way that the status API works in GitHub it will hide all but the latest run with a specific context name, making it hard fo figure out where to find all the results. This could be solved by giving the ability to override the context name for a specific set of tasks from the `.taskcluster.yml` file.

The main issue is presumably that this wants to be per repository, not per task, so one ends up duplicating some of the task logic to set the right context for each event.

**Comment by djmitche** [comment]
I suspect the checks API might be helpful here, too -- those different groups could create different checks?

**Comment by jgraham** [comment]
Yeah, I think that integrating with the checks API would be another way of addressing this use case.

**Comment by djmitche** [comment]
That's in https://bugzilla.mozilla.org/show_bug.cgi?id=1459645 if you want
to follow along (noting it's not currently scheduled to be worked on)

**Comment by SimonSapin** [comment]
This will likely also affect Servo.

When a PR is approved by a reviewer, our bot makes a merge commit for it and pushes to the `auto` branch. We configure `.taskcluster.yml` to start testing tasks for this push. If tests are green, that merge commit is then pushed to `master`.

Now I’m considering adding tasks that start on that second push. (Namely, uploading to https://doc.servo.org/ , only when we know a PR is being merged.) As far as I understand this new status will overwrite the previous one in GitHub’s web UI. (All data is still there in the API, though.)

**Comment by djmitche** [comment]
That, too, may be a good reason to use the checks API.  GitHub's history suggests that the status API will slowly fall into disrepair, so adding additional functionality to it is probably not a good investment.
